src/map/latent_effect_container.cpp

```diff
@@ -788,9 +788,9 @@ bool CLatentEffectContainer::ProcessLatentEffect(CLatentEffect& latentEffect)
         {
             size_t partyCount = 0;
             size_t trustCount = 0;
-            if (m_POwner->PParty != nullptr)
+            if (m_POwner->PParty && dynamic_cast<CCharEntity*>(m_POwner->PParty->GetLeader()))
```

Comment:
For both `case LATENT::PARTY_MEMBERS:` and `case LATENT::PARTY_MEMBERS_IN_ZONE:`; doing this `dynamic_cast` and throwing away the result to then do a `static_cast` to get that same result seems very off. Since you only really want to continue if there is a leader, extract out the leader first.

```cpp
            size_t partyCount = 0;
            size_t trustCount = 0;
            auto* PLeader = m_POwner->PParty != nullptr ?  dynamic_cast<CCharEntity*>(m_POwner->PParty->GetLeader()) : nullptr;
            if (PLeader) ...
 ....